third_party/terraform/tests/data_source_iam_beta_workload_identity_pool_test.go.erb

```diff
@@ -0,0 +1,47 @@
+<% autogen_exception -%>
+package google
+
+<% unless version == 'ga' -%>
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceIAMBetaWorkloadIdentityPool_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIAMBetaWorkloadIdentityPoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceIAMBetaWorkloadIdentityPoolBasic(context),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceState("data.google_iam_workload_identity_pool.foo", "google_iam_workload_identity_pool.bar"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceIAMBetaWorkloadIdentityPoolBasic(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_iam_workload_identity_pool" "bar" {
+	workload_identity_pool_id = "bar-pool-%{random_suffix}"
+	display_name              = "Name of pool"
+	description               = "Identity pool for automated test"
+	disabled                  = true
+}
+
+data "google_iam_workload_identity_pool" "foo" {
+	workload_identity_pool_id = google_iam_workload_identity_pool.bar.workload_identity_pool_id
```

Comment:
Out of curiosity, does this work if you use `google_iam_workload_identity_pool.bar.id` here? Generally our team prefers to try to refer to the id of a resource instead of a property on it when setting up a reference to that resource as a whole (https://github.com/hashicorp/terraform-provider-google/wiki/Developer-Best-Practices#self-links-names-and-ids). It's not a hard requirement though, so if it's too difficult to get working then we can skip it.

Reply:
Not without making changes. Referencing the `id` or `name` both give the full name of the identity pool in the format `projects/{{project}}/locations/global/workloadIdentityPools/{{workload_identity_pool_id}}`. The `workload_identity_pool_id` attribute we are specifying here needs only the simple name of the pool. That was supplied to that pool as `workload_identity_pool_id` (a URL only parameter) during creation.